schemachange/sc_logic.c

```diff
@@ -1138,9 +1170,23 @@ int resume_schema_change(void)
                     tpt_sc->s = s;
                     tpt_sc->nshards++;
                 }
-            } else if (s->finalize == 0 && s->kind != SC_ALTERTABLE_PENDING) {
-                s->sc_next = sc_resuming;
-                sc_resuming = s;
+            } else {
+                /* start the schema change back up */
+                rc = start_schema_change(s);
+                if (rc != SC_OK && rc != SC_ASYNC) {
+                    logmsg(
+                            LOGMSG_ERROR,
+                            "%s: failed to resume schema change for table '%s' rc %d\n",
+                            __func__, s->tablename, rc);
+                    /* start_schema_change will free if this fails */
+                    /*
+                       free_schema_change_type(s);
+                       */
+                    continue;
+                } else if (s->finalize == 0 && s->kind != SC_ALTERTABLE_PENDING) {
+                    s->sc_next = sc_resuming;
+                    sc_resuming = s;
+                }
             }
         }
     }
```

Comment:
Are there any existing tests for resuming partitioned schema changes?

Reply:
A few tests swing the master during schema changes, sc_resume is dedicated for that